dspace-spring-rest/src/main/java/org/dspace/app/rest/converter/EPersonConverter.java

```diff
@@ -40,10 +47,18 @@ public EPersonRest fromModel(EPerson obj) {
         eperson.setRequireCertificate(obj.getRequireCertificate());
         eperson.setSelfRegistered(obj.getSelfRegistered());
         eperson.setEmail(obj.getEmail());
+
+        return eperson;
+    }
+
+    public EPersonRest fromModelWithGroups(Context context, EPerson ePerson) throws SQLException {
```

Comment:
Needs JavaDocs.

Also, a simple question here, why do you not just update the existing `EPersonConverter.fromModel()` method with the correct way to obtain all member groups?  I'm not sure I understand why we are changing the behavior of `fromModel()` to no longer include the EPerson's group information, and then adding a new method `fromModelWithGroups()`?   (It's very possible that this could be cleared up via JavaDocs though...I just might not be seeing the full picture here.)

Reply:
Hi @tdonohue, some repositories contain many (thousands) of groups and some epersons are member of many (hundreds) of groups. If we would include the groups in the regular `fromModel` method, that would mean that the `/api/eperson/epersons` (which lists all available groups) response would become huge and slow.

The moment we support REST model projects, we could merge these two methods again. But for now, I would leave them separate. I'll add some JavaDoc to the method.